### ISSUE TITLE
Connection problems since ESPhome 2021.08.0 upgrade

### ISSUE THREAD

**Issue by FreddyLauwers** [esphome/issues#2330]
### The problem

Hi,
I have a connection problem with several ESPhome devices since the ESPhome 2021.8.0 update from yesterday.
All devices are pingable on the network without packet loss and sometimes are able to connect for a while.
For example my Sonof rf-bridge, several D1 Mini's and some Blitzwolf BW-SHP6 all have this problem since the upgrade with no code changes on the devices, just recompilation for the new version.
Thank you for your work!

### Which version of ESPHome has the issue?

2021.08

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2021.8.8

### What platform are you using?

ESP8266

### Anything in the logs that might be useful for us?

```txt
Log example from Sonoff rf-bridge, but the same on several other devices that were upgraded.

[11:16:38][C][api:095]: API Server:
[11:16:38][C][api:096]:   Address: 172.28.3.3:6053
[11:16:38][C][wifi_signal.sensor:009]: WiFi Signal 'Sonoff RF Bridge Wifi Signal'
[11:16:38][C][wifi_signal.sensor:009]:   Device Class: 'signal_strength'
[11:16:38][C][wifi_signal.sensor:009]:   State Class: 'measurement'
[11:16:38][C][wifi_signal.sensor:009]:   Unit of Measurement: 'dBm'
[11:16:38][C][wifi_signal.sensor:009]:   Accuracy Decimals: 0
[11:16:43][D][sensor:131]: 'Sonoff RF Bridge Wifi Signal': Sending state -43.00000 dBm with 0 decimals of accuracy
[11:16:53][D][sensor:131]: 'Sonoff RF Bridge Wifi Signal': Sending state -43.00000 dBm with 0 decimals of accuracy
[11:17:03][D][sensor:131]: 'Sonoff RF Bridge Wifi Signal': Sending state -44.00000 dBm with 0 decimals of accuracy
WARNING Disconnected from API: Timeout while waiting for message response!
INFO Connecting to 172.28.3.3:6053 (172.28.3.3)
WARNING Couldn't connect to API (Error connecting to 172.28.3.3: timed out). Trying to reconnect in 1 seconds
INFO Connecting to 172.28.3.3:6053 (172.28.3.3)
WARNING Couldn't connect to API (Error connecting to 172.28.3.3: timed out). Trying to reconnect in 1 seconds
INFO Connecting to 172.28.3.3:6053 (172.28.3.3)
WARNING Couldn't connect to API (Error connecting to 172.28.3.3: [Errno 111] Connection refused). Trying to reconnect in 2 seconds
INFO Connecting to 172.28.3.3:6053 (172.28.3.3)
WARNING Couldn't connect to API (Error connecting to 172.28.3.3: [Errno 111] Connection refused). Trying to reconnect in 3 seconds
INFO Connecting to 172.28.3.3:6053 (172.28.3.3)
WARNING Couldn't connect to API (Error connecting to 172.28.3.3: [Errno 111] Connection refused). Trying to reconnect in 5 seconds
INFO Connecting to 172.28.3.3:6053 (172.28.3.3)
WARNING Couldn't connect to API (Error connecting to 172.28.3.3: [Errno 111] Connection refused). Trying to reconnect in 7 seconds
INFO Connecting to 172.28.3.3:6053 (172.28.3.3)
WARNING Couldn't connect to API (Error connecting to 172.28.3.3: [Errno 111] Connection refused). Trying to reconnect in 11 seconds


Other device with similar problems:

[19:19:12][C][homeassistant.time:010]: Home Assistant Time:
[19:19:12][C][homeassistant.time:011]:   Timezone: 'CET-1CEST-2,M3.4.0/2,M10.5.0/3'
[19:19:12][C][version.text_sensor:021]: Version Text Sensor 'comp_1_ESPHome_Version'
ERROR Error while reading incoming messages: Error while receiving data: [Errno 104] Connection reset by peer
WARNING Disconnected from API: Error while receiving data: [Errno 104] Connection reset by peer
INFO Connecting to 172.28.3.14:6053 (172.28.3.14)
WARNING Couldn't connect to API (Timeout while waiting for message response!). Trying to reconnect in 1 seconds
INFO Connecting to 172.28.3.14:6053 (172.28.3.14)
INFO Successfully connected to 172.28.3.14
[19:19:31][D][sensor:131]: 'comp_1_Wattage': Sending state 52.46222 W with 1 decimals of accuracy
[19:19:31][D][sensor:131]: 'comp_1_Energy': Sending state 0.14573 Wh with 1 decimals of accuracy
WARNING Disconnected from API: Timeout while waiting for message response!
INFO Connecting to 172.28.3.14:6053 (172.28.3.14)
INFO Successfully connected to 172.28.3.14
[19:19:38][D][api.connection:730]: Client 'Home Assistant 2021.8.8 (172.28.0.4)' connected successfully!
ERROR Error while reading incoming messages: Error while receiving data: [Errno 104] Connection reset by peer
WARNING Disconnected from API: Error while receiving data: [Errno 104] Connection reset by peer
INFO Connecting to 172.28.3.14:6053 (172.28.3.14)
WARNING Couldn't connect to API (Timeout while waiting for message response!). Trying to reconnect in 1 seconds
INFO Connecting to 172.28.3.14:6053 (172.28.3.14)
INFO Successfully connected to 172.28.3.14
[19:19:56][D][sensor:131]: 'comp_1_Wattage': Sending state 52.60635 W with 1 decimals of accuracy
[19:19:56][D][sensor:131]: 'comp_1_Energy': Sending state 0.14613 Wh with 1 decimals of accuracy
[19:20:06][D][hlw8012:082]: Got power=53.3W, voltage=243.6V
[19:20:06][D][sensor:131]: 'comp_1_Voltage': Sending state 243.55676 V with 1 decimals of accuracy
[19:20:06][D][sensor:131]: 'comp_1_Wattage': Sending state 53.32698 W with 1 decimals of accuracy
[19:20:06][D][sensor:131]: 'comp_1_Energy': Sending state 0.29426 Wh with 1 decimals of accuracy
[19:20:16][D][hlw8012:082]: Got power=52.2W, voltage=243.5V
[19:20:16][D][sensor:131]: 'comp_1_Voltage': Sending state 243.47420 V with 1 decimals of accuracy
[19:20:16][D][sensor:131]: 'comp_1_Wattage': Sending state 52.17397 W with 1 decimals of accuracy
[19:20:16][D][sensor:131]: 'comp_1_Energy': Sending state 0.43919 Wh with 1 decimals of accuracy
[19:20:18][D][sensor:131]: 'comp_1_Uptime': Sending state 0.01061 Hrs with 1 decimals of accuracy
[19:20:26][D][hlw8012:082]: Got power=52.8W, voltage=243.6V
[19:20:26][D][sensor:131]: 'comp_1_Voltage': Sending state 243.58977 V with 1 decimals of accuracy
[19:20:26][D][sensor:131]: 'comp_1_Wattage': Sending state 52.75047 W with 1 decimals of accuracy
[19:20:26][D][sensor:131]: 'comp_1_Energy': Sending state 0.58572 Wh with 1 decimals of accuracy



```

**Comment by FreddyLauwers** [comment]
An additional observation that may help is when I change the setting 'status_use_ping' to enabled in the configuration, the devices seem to lose the API connection less frequent.

**Comment by FreddyLauwers** [comment]
In the meantime, I have reinstalled HA from scratch and then restored my latest backup and everything works just fine, every time. All ESPHome devices came online immediately and stayed online… But I hadn’t configured my IoT VLAN yet and I let the devices connect through a firewall rule to the main adapter. After I configured my IoT VLAN on HA and let the device connect to the interface on this VLAN, problems came back immediately. So in my humble opinion, this has something to do with ESPHome and VLANs. Maybe this is also an explanation why most people don’t experience this problem if they use everything on the same lan.
